dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/export/trackedentity/DefaultTrackedEntityService.java

```diff
@@ -240,10 +236,11 @@ public Page<TrackedEntity> findTrackedEntities(
     final Page<TrackedEntityIdentifiers> ids =
         trackedEntityStore.getTrackedEntityIds(queryParams, pageParams);
 
-    List<TrackedEntity> trackedEntities =
-        findTrackedEntities(ids.getItems(), operationParams, queryParams, user);
+    Map<String, TrackedEntity> trackedEntities =
+        findTrackedEntities(ids.getItems(), operationParams, queryParams, user).stream()
+            .collect(Collectors.toMap(TrackedEntity::getUid, Function.identity()));
 
-    return ids.withFilteredItems(trackedEntities);
+    return ids.withMappedItems(id -> trackedEntities.get(id.uid()));
```

Comment:
Now that we are mapping items, can we remove the ordering from the aggregate query?
In my head it makes sense that pagination and order is defined in the ids query and then the aggregate is just getting more data and do not need to worry on order.

So it would be removing the call to `applySortOrder` [here](https://github.com/dhis2/dhis2-core/blob/ba958355873f0eadac98a1824a02371a688792e4/dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/export/trackedentity/aggregates/TrackedEntityStore.java#L86)

Reply:
We could still map from ids to tracked entity and make findTrackedEntities unordered.
It should be faster to map than order in the DB.